valohai_yaml/pipelines/conversion.py

```diff
@@ -62,6 +62,11 @@ def convert_executionlike_node(self, node: Union[ExecutionNode, TaskNode]) -> Co
         step_data = step.serialize()
         step_data.update(override)
 
+        runtime_config = step_data.get("runtime_config", {})
+        if 'no-output-timeout' in step_data:
+            runtime_config.setdefault('no_output_timeout', step_data.pop('no-output-timeout'))
+        step_data["runtime_config"] = runtime_config
```

Comment:
Could simplify with `setdefault`:
```suggestion
        runtime_config = step_data.setdefault("runtime_config", {})
        if "no-output-timeout" in step_data:
            runtime_config.setdefault("no_output_timeout", step_data.pop("no-output-timeout"))
```